widget/embedded/src/pages/ConfirmSwapPage.tsx

```diff
@@ -65,55 +63,64 @@ export function ConfirmSwapPage() {
   //TODO: move component's logics to a custom hook
   const {
     selectedQuote,
-    setInputAmount,
-    selectedWallets,
-    quoteWalletsConfirmed,
-    customDestination,
+    inputAmount,
     quoteWarningsConfirmed,
+    confirmSwapData,
+    customDestination,
+    setInputAmount,
   } = useQuoteStore();
+  const { disabledLiquiditySources, customSlippage, slippage } = useAppStore();
+  const tokens = useAppStore().tokens();
+  const blockchains = useAppStore().blockchains();
+  const sourceWallet = useAppStore().selectedWallet('source');
+  const destinationWallet = useAppStore().selectedWallet('destination');
+  const { isActiveTab } = useUiStore();
   const navigate = useNavigate();
+  const { manager } = useManager();
   const [dbErrorMessage, setDbErrorMessage] = useState<string>('');
-
-  const showWalletsOnInit = !quoteWalletsConfirmed;
-  const [showWallets, setShowWallets] = useState(false);
   const [isConfirming, setIsConfirming] = useState(false);
-  const { isActiveTab } = useUiStore();
-  const disabledLiquiditySources = useAppStore().getDisabledLiquiditySources();
   const prevDisabledLiquiditySources = useRef(disabledLiquiditySources);
-  const { manager } = useManager();
   const {
     fetch: confirmSwap,
     loading: fetchingConfirmationQuote,
     cancelFetch,
   } = useConfirmSwap();
   const [confirmSwapResult, setConfirmSwapResult] =
     useState<ConfirmSwapFetchResult>({
-      swap: null,
+      response: confirmSwapData.result,
       error: null,
       warnings: null,
     });
   const [showQuoteWarningModal, setShowQuoteWarningModal] = useState(false);
-
-  const onConfirmSwap: ConfirmSwap['fetch'] = async ({
-    selectedWallets,
-    customDestination,
-  }) => {
-    const result = await confirmSwap?.({ selectedWallets, customDestination });
-
-    setConfirmSwapResult(result);
-    return result;
-  };
+  const selectedWalletsForConfirmation = sourceWallet
+    ? [sourceWallet].concat(destinationWallet ?? [])
+    : [];
```

Comment:
Can we use this directly, or do we need to ensure that the destination wallet is only included if the source wallet exists?

```suggestion
[sourceWallet, destinationWallet].filter(Boolean);
```

If we want to guarantee that the source wallet is present before including anything, then this would be a nit:
```suggestion
const selectedWalletsForConfirmation = sourceWallet
  ? [sourceWallet, destinationWallet].filter(Boolean)
  : [];
```